botocore/endpoint_provider.py

```diff
@@ -600,7 +601,7 @@ def __init__(
         except AttributeError:
             raise EndpointResolutionError(
                 msg=f"Unknown parameter type: {parameter_type}. "
-                "A parameter must be of type string or boolean."
+                "A parameter must be of type string, boolean, or stringarray."
```

Comment:
`stringarray` isn't a type in python. We should probably instead say something like `list[str]` which lets the user know they should use a list of strings.

Reply:
`stringarray` isn't a type, but we have added it as an enum value in the class `ParameterType` which is a "translation from `type` attribute to native Python type."

Reply:
This is the definition of a parameter, where there are custom, language-agnostic types.  This is defined in the ruleset and not by the end-users; here we are checking the name of the parameter type.

We are only checking the name of the parameter that's defined in the ruleset, for instance as it is shown in the test below:

```
  "parameters": {
    "ResourceList": {
      "type": "stringArray"
    }
  }
```

The valid types are `string`, `boolean`, and `stringArray`.  All of these get lowercased on line 599, so I've kept it lowercase in the error message, but I'm happy to add back in the uppercase or modify the message as you see fit.  

Based on this, what do you think is the right formatting for the message?

Reply:
Thank you both for the explanation. Let's leave this as `stringarray`.